pkg/sentry/server/ca/selfhosted.go

```diff
@@ -18,60 +18,138 @@ import (
+	return bundle, ca_bundle.MissingCredentials{
+		X509: missingX509,
+		JWT:  missingJWT,
+	}, nil
+}
+
+// loadAndValidateX509Bundle loads the X.509 certificates and keys from disk, verifies them, and updates the bundle. Returns whether any are missing.
+func (s *selfhosted) loadAndValidateX509Bundle(bundle *ca_bundle.Bundle) (bool, error) {
+	missingX509 := false
+
+	// Read trust anchors (root certificate)
+	trustAnchors, err := os.ReadFile(s.config.RootCertPath)
 	if os.IsNotExist(err) {
-		return Bundle{}, false, nil
+		missingX509 = true
+	} else if err != nil {
+		return false, fmt.Errorf("failed to read root certificate: %w", err)
 	}
-	if err != nil {
-		return Bundle{}, false, err
+
+	// Read issuer certificate chain
+	issChainPEM, err := os.ReadFile(s.config.IssuerCertPath)
+	if os.IsNotExist(err) {
+		missingX509 = true
+	} else if err != nil {
+		return false, fmt.Errorf("failed to read issuer certificate: %w", err)
 	}
 
+	// Read issuer private key
 	issKeyPEM, err := os.ReadFile(s.config.IssuerKeyPath)
 	if os.IsNotExist(err) {
-		return Bundle{}, false, nil
+		missingX509 = true
+	} else if err != nil {
+		return false, fmt.Errorf("failed to read issuer key: %w", err)
 	}
-	if err != nil {
-		return Bundle{}, false, err
+
+	if !missingX509 {
+		verifiedBundle, err := verifyBundle(trustAnchors, issChainPEM, issKeyPEM)
+		if err != nil {
+			return false, fmt.Errorf("failed to verify CA bundle: %w", err)
+		}
+		*bundle = verifiedBundle
 	}
 
-	bundle, err := verifyBundle(trustAnchors, issChainPEM, issKeyPEM)
-	if err != nil {
-		return Bundle{}, false, fmt.Errorf("failed to verify CA bundle: %w", err)
+	return missingX509, nil
+}
+
+// loadAndValidateJWTBundle loads the JWT signing key and JWKS from disk, verifies them, and updates the bundle. Returns whether any JWT credentials are missing.
+func (s *selfhosted) loadAndValidateJWTBundle(bundle *ca_bundle.Bundle) (bool, error) {
+	missingJWT := false
+
+	// Read JWT signing key
+	jwtKeyPEM, err := os.ReadFile(s.config.JWT.SigningKeyPath)
+	if err == nil {
+		jwtKey, jwtErr := loadJWTSigningKey(jwtKeyPEM)
+		if jwtErr != nil {
+			return false, fmt.Errorf("failed to load JWT signing key: %w", jwtErr)
+		}
+		bundle.JWT.SigningKey = jwtKey
+		bundle.JWT.SigningKeyPEM = jwtKeyPEM
+	} else if os.IsNotExist(err) {
+		missingJWT = true
+	} else {
+		return false, fmt.Errorf("error reading JWT signing key: %w", err)
+	}
+
+	// Read JWKS
+	jwks, err := os.ReadFile(s.config.JWT.JWKSPath)
+	if err == nil {
+		if verifyErr := verifyJWKS(jwks, bundle.JWT.SigningKey, s.config.JWT.KeyID); verifyErr != nil {
+			return false, fmt.Errorf("failed to verify JWKS: %w", verifyErr)
+		}
+		bundle.JWT.JWKSJson = jwks
+		bundle.JWT.JWKS, err = jwk.Parse(jwks)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse JWKS: %w", err)
+		}
+	} else if os.IsNotExist(err) {
+		missingJWT = true
+	} else {
+		return false, fmt.Errorf("error reading JWKS: %w", err)
```

Comment:
can this be tweaked a little bc I think it could be a bit cleaner -
What if we move the stuff from in the `if err == nil` outside of a conditional, and we just do `if err != nil { return false, fmt.Errorf("error reading JWKS: %w", err)}` and `else if os.IsNotExist(err) {missingJWT = true}`

Hopefully that makes sense, otherwise I can rephrase it. lmk :)